lib/babe/crypto_test.go

```diff
@@ -1,4 +1,187 @@
 // Copyright 2021 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package babe
\ No newline at end of file
+package babe
+
+import (
+	"errors"
+
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/stretchr/testify/assert"
+
+	"testing"
+)
+
+func TestCalculateThreshold(t *testing.T) {
+	type args struct {
+		C1       uint64
+		C2       uint64
+		numAuths int
+	}
+	tests := []struct {
+		name   string
+		args   args
+		exp    *scale.Uint128
+		expErr error
+	}{
+		{
+			name: "happy path",
+			args: args{
+				C1:       1,
+				C2:       2,
+				numAuths: 3,
+			},
+			exp: &scale.Uint128{Upper: 0x34d00ad6148e1800, Lower: 0x0},
+		},
+		{
+			name: "0 value input",
+			args: args{
+				C1:       0,
+				C2:       0,
+				numAuths: 0,
+			},
+			expErr: ErrThresholdBothZero,
+		},
+		{
+			name: "C1 > C2",
+			args: args{
+				C1:       5,
+				C2:       2,
+				numAuths: 0,
+			},
+			expErr: errors.New("invalid C1/C2: greater than 1"),
+		},
+		{
+			name: "max threshold",
+			args: args{
+				C1:       2147483647,
+				C2:       2147483647,
+				numAuths: 3,
```

Comment:
```suggestion
			args: args{
				C1:       1,
				C2:       1,
				numAuths: 3,
```

can you do this (since it's the c1/c2 fraction that gets used in the equation)?

Reply:
Yup good call much cleaner